web/src/stream_topic_history.js

```diff
@@ -138,6 +138,9 @@ export class PerStreamHistory {
 
         if (!existing.historical) {
             existing.count += 1;
+        } else {
+            existing.historical = false;
+            existing.count = 1;
```

Comment:
Does this code know that the message we're adding is "new", and not something we're fetching via a backfill? I think the answer is yes, but it's not been easy for me to prove.